orca-spring-batch/src/main/java/com/netflix/spinnaker/orca/batch/stages/SpringBatchStageBuilderProvider.java

```diff
@@ -37,7 +38,16 @@ public SpringBatchStageBuilderProvider(ApplicationContext applicationContext,
     this.stageBuilders.addAll(
       stageDefinitionBuilders
       .stream()
-      .map(s -> new LinearStageDefinitionBuilder(s, this))
+      .map(s -> {
+        if (s instanceof BranchingStageDefinitionBuilder) {
+          BranchingStageDefinitionBuilder branchingStageDefinitionBuilder = (BranchingStageDefinitionBuilder) s;
+          if (s.getType().equals("deploy")) {
```

Comment:
This is a bit hokey ... but deploy has very specific logic for building flows vs. other parallel stages.

Happy to chat about a better solution (I'll at least change it so it's not comparing against "deploy").